Apps/Src/Amps.c

```diff
@@ -54,13 +74,65 @@ SafetyStatus Amps_CheckStatus(bool chargingOnly) {
  * @return 1 if charge, 0 if discharge
  */
 int8_t Amps_IsCharging(void) {
-	return latestMeasureMilliAmps < 0;
+	// TODO: Make sure that the amperes board is installed in such a way that negative => charging
+	OSMutexPend(&AmperesData_Mutex, 0, OS_OPT_PEND_BLOCKING, &ticks, &err);
```

Comment:
There is no race condition here. No need for a lock.

Reply:
I will fix